Wikipedia/Code/YearInReviewSettingsViewController.swift

```diff
@@ -105,7 +106,15 @@ final class YearInReviewSettingsViewController: SubSettingsViewController {
 
         Task {
             do {
-                let yirDataController = try WMFYearInReviewDataController()
+                
+                var userId: Int?
+
+                if let siteURL = dataStore.languageLinkController.appLanguage?.siteURL,
+                   let userID = dataStore.authenticationManager.permanentUser(siteURL: siteURL)?.userID {
+                    userId = userID
+                }
+                
+                let yirDataController = try WMFYearInReviewDataController(userID: userId)
```

Comment:
Here you'll pass `userID` into the next line as a parameter instead of into the init.